### ISSUE TITLE
[F6] x/liquidation: preCheckCollateral 对 cross 用 Collateral 而非 TAV

### ISSUE THREAD

**Issue by lyqingye** [lyqingye/perpdex-l1#78]
## 背景

ADL deleverager 候选必须有足够 cross collateral 才能吸收 fill。`preCheckCollateral` 检查候选 collateral 是否覆盖本笔 realized loss。

[x/liquidation/keeper/liquidate.go](x/liquidation/keeper/liquidate.go) L377-386：

```go
switch pos.MarginMode {
case perptypes.IsolatedMargin:
    available = pos.AllocatedMargin
default:
    a, err := k.accountKeeper.GetAccount(ctx, accountIdx)
    available = a.Collateral  // ← 仅读"已实现现金", 不含 uPnL
}
```

跟踪自 epic #65 的 F6。

## 问题分析

cross 模式下：
- `account.Collateral` = 已结算的 USDC 现金
- `TAV` = `Collateral + Σ uPnL_per_market`

对 ADL profitable winner 候选——他们之所以入选就是因为有大量正 uPnL——这两个值差距可能很大。

### 数值演示

Bob 账户：
```
collateral       = 1,000 USDC
ETH long uPnL    = +5,000
BTC short uPnL   = +3,000
TAV              = 9,000

本次 ADL 让他平掉部分 ETH long, realized loss = -2,500
required cushion = 2,500
available (现行 Collateral) = 1,000  ← 不够, skip
available (TAV 口径)        = 9,000  ← 够, 应该可以吃
```

Bob TAV 充足、整体健康，却因为只看 `Collateral` 字段被错误跳过。

## 进一步分析

该分支只在 `realized.IsNegative()` 时走（[liquidate.go](x/liquidation/keeper/liquidate.go) L367-374 short-circuit）。对 ADL profitable winner：

- close 操作把 uPnL 锁定为 realized profit，realized 通常为正 → 短路 return
- 但 funding 翻转 / ZP 漂移导致这笔 close 是亏损时（realized < 0），才会走到这个分支

## 影响

- 概率较低
- 发生时：候选被错误 skip → ADL 吞吐下降 → 残债概率上升
- 可能掩盖 spec 不一致

## 修复建议

- [ ] 与 Lighter spec 对照确认 `is_deleverager_has_enough_cross_collateral` 应该用哪个口径
- [ ] **方案 A**：保持 `Collateral`，加注释解释"故意保守以与 spec 对齐"
- [ ] **方案 B**：改用 TAV-like 口径，公式：

```go
// 排除当前正在 close 的市场，避免重复计算
available = a.Collateral + Σ uPnL_for_other_markets
```

## 参考

- Epic: #65
- [todo/liquidation_codereview/01_codereview.md](todo/liquidation_codereview/01_codereview.md) F6
